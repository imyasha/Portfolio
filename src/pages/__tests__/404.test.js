import React from 'react';
import * as Gatsby from 'gatsby';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import ErrorPage from '../404';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      description: '',
      title: '404 page',
      lang: 'en',
      meta: [],
      keywords: [],
    },
  },
}));

test('renders correctly', () => {
  const { container } = render(<ErrorPage />);
  expect(container.firstChild).toMatchSnapshot();
});
