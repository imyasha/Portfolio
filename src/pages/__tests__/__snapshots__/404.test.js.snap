// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  width: 1140px;
  max-width: 94%;
  margin: 0 auto;
}

<div
  class="c0"
>
  <h1>
    This page was not found
  </h1>
  <p>
    I'm sorry for the confusion. Please go back and try again.
  </p>
</div>
`;
